### DISCUSSION TITLE
How to draw curved lines to make them look rounded

### DISCUSSION THREAD

**Discussion by bibiehy** [pixijs/pixijs#8561] [Q&A]
pixi@6.5.1

Suppose there are 100 coordinate points, connected by lines, it looks like this, how to do it through pixi.
https://codepen.io/bibiehy/pen/abYKowB?editors=0011
![image](https://user-images.githubusercontent.com/35096473/183294119-3bbd3258-2a04-4cae-b5e8-e0dea53d43f4.png)

**Comment by Julien-Marcou** [comment]
There are several ways to achieve this, but one that can be easily understood, is by computing the tangent of the curve at each points and use the tangents to draw bezier curves.

![Tangent example](https://user-images.githubusercontent.com/4369075/183399014-ec32e91f-ed77-4aed-86d8-a84151e8f690.png)

You can find more information on Stack Overflow:
https://stackoverflow.com/questions/7054272/how-to-draw-smooth-curve-through-n-points-using-javascript-html5-canvas

There is a nice demo where you can click on the canvas to add points and see how it works:
https://stackoverflow.com/a/20309900/2837434

Usualy, this implementation requires a "tension" parameter that ranges from 0 to 1, which makes the tangents shorter or longer, with 0 you get straight lines, with 1 you get extravagant curves, so you most likely want a tension of 0.5

**Reply by bibiehy** [reply]
really great, thank you
